File: src/main/java/sage/transfer/TopicReplyView.java
package sage.transfer;

import java.sql.Timestamp;

import sage.entity.TopicReply;

public class TopicReplyView {
  public Long id;
  public String content;
  public Long topicPostId;
  public UserLabel author;
  public Timestamp whenCreated;
  /** Nullable */
  public UserLabel toUser;
  /** Nullable */
  public Long toReplyId;

  TopicReplyView() {}

  public TopicReplyView(TopicReply reply, UserLabel toUserLabel) {
    id = reply.getId();
    content = reply.getContent();
    topicPostId = reply.getTopicPostId();
    author = new UserLabel(reply.getAuthor());
    whenCreated = reply.getWhenCreated();
    toUser = toUserLabel;
    toReplyId = reply.getToReplyId();
  }
}

File: src/main/kotlin/sage/transfer/TopicReplyView.kt
package sage.transfer

import sage.annotation.KotlinNoArg
import java.sql.Timestamp

import sage.entity.TopicReply

@KotlinNoArg
class TopicReplyView {
  var id: Long = 0
  var content: String = ""
  var topicPostId: Long = 0
  var author: UserLabel? = null
  var whenCreated: Timestamp? = null
  var floorNumber: Int = 0

  var toUser: UserLabel? = null
  var toReplyId: Long? = null

  constructor(reply: TopicReply, toUserLabel: UserLabel?) {
    id = reply.id
    content = reply.content
    topicPostId = reply.topicPostId
    author = UserLabel(reply.author)
    whenCreated = reply.whenCreated
    floorNumber = reply.floorNumber

    toUser = toUserLabel
    toReplyId = reply.toReplyId
  }
}
